### DISCUSSION TITLE
ModelState validação automática

### DISCUSSION THREAD

**Discussion by CVieiraSantos** [balta-io/forum#29] [Tira Dúvidas] [answered]
@andrebaltieri  Balta tenho uma dúvida. Quando você mostrou os erros e o status Ok com a validação automática, essa mesma validação automática vem daquele ModelState.IsValid? Me refiro aquelas mensagens em inglês que apresenta no Portman quando  o Docker estava fora? É isso que o ModelState.IsValid? faz por baixo dos panos quando não colocamos ele ou quando colocamos?

Outra pergunta. Quando mudamos o comportamento da API por meio do ConfigureApiBehaviorOptions e ai somos obrigados a colocar o ModelState.IsValid é obrigação configurar essa validação e por isso foi inserida a classe ResultViewModel no retorno da API? O objetivo seria para mudar a mensagem que é apresentada para o usuário, ou seja, para entender o que está acontecendo?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @CVieiraSantos, como vai?

No caso, as validações feitas "automaticamente" quando usamos o atributo `[ApiController]` são sim feitas pelo ModelState. Porém, o erro que você visualiza quando o Docker está fora é o tratamento de erro global do ASP.NET. Como estamos em modo de debug ele dá estas instruções na tela para ajudar a encontrar o erro.

Sobre sua outra pergunta, sim, se você quer mudar o comportamento, precisa deixar de utilizar o `[ApiController]` e passar a usar as validações como mostro no curso de ASP.NET 6.

**Comment by CVieiraSantos** [comment]
Boa noite Balta.
@andrebaltieri só mais uma dúvida. Porque aquele linha if (!ModelState.IsValid)
                return BadRequest(new ResultViewModel<Category>(ModelState.GetErrors())); só é testado no método Post e não é testado nos demais, e principalmente no Put?
Ou seja, porque o ModelState.IsValid só é inserido no Post, uma vez que o Put também tem um [FromBody] no copo do método?

**Reply by andrebaltieri** [reply]
@CVieiraSantos foi um exemplo... você **DEVE** validar no PUT e demais métodos que tenham interações de dados.
